pkg/provisioner/ironic/ironic.go

```diff
@@ -1820,6 +1846,8 @@ func (p *ironicProvisioner) softPowerOff() (result provisioner.Result, err error
 		// If the target state is unset while the last error is set,
 		// then the last execution of soft power off has failed.
 		if targetState == "" && ironicNode.LastError != "" {
+			p.log.Info("soft power off error", "msg", ironicNode.LastError)
+			result, _ = operationFailed(ironicNode.LastError)
 			return result, SoftPowerOffFailed{}
```

Comment:
Returning the failure as an `error` here instead of putting it directly in `result.ErrorMessage` in inconsistent with every other method that returns (`provisioner.Result, error)`, which we then have to reverse in the caller.
Let's take this opportunity to `return operationFailed(ironicNode.LastError)`.

Reply:
Fixed.